src/monmove.cpp

```diff
@@ -1774,11 +1775,17 @@ bool monster::attack_at( const tripoint &p )
 
 static tripoint find_closest_stair( const tripoint &near_this, const ter_furn_flag stair_type )
 {
-    map &here = get_map();
-    for( const tripoint &candidate : closest_points_first( near_this, 10 ) ) {
+    const map &here = get_map();
+    std::optional<tripoint> candidate = find_point_closest_first( near_this, 10, [&here,
+    stair_type]( const tripoint & candidate ) {
         if( here.has_flag( stair_type, candidate ) ) {
-            return candidate;
+            return true;
         }
```

Comment:
```suggestion
        } else {
            return false;
```
that supposed to fix it iirc

Reply:
Yes, that would do it. Thanks for finding that.